### DISCUSSION TITLE
figure out a prioritised list of WebAPIs to work on first post-RFC

### DISCUSSION THREAD

**Discussion by kelset** [microsoft/rnx-kit#4342] [Ideas]
As part of the work on this effort (see [the RFC for details](https://github.com/microsoft/rnx-kit/pull/2504)), we need to prioritize from the massive list of WebAPIs we use (the one we generate with https://github.com/microsoft/rnx-kit/tree/main/incubator/%40react-native-webapis/scan and collect [in this file](https://github.com/microsoft/rnx-kit/blob/main/incubator/%40react-native-webapis/scan/merge.mjs)) some that are more important than others, and basically create a shortlist of 10-ish that we can use as the first pool of work.

The file gives us a data-driven perspective, so we'll leverage it to decide what to work on. If you want to help out and work on some webAPIs not in the current list, please leave a comment here!

Current list of APIs:

- [x] [Battery Status API](https://developer.mozilla.org/en-US/docs/Web/API/Battery_Status_API) (@tido64)
- [ ] [Payment Handler API 🧪](https://developer.mozilla.org/en-US/docs/Web/API/Payment_Handler_API) (@matthargett?)
- [ ] [Payment Request API](https://developer.mozilla.org/en-US/docs/Web/API/Payment_Request_API) (@matthargett?)
- [x] [Web Storage API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API) (@tido64)
- [ ] [WebGPU 🧪](https://developer.mozilla.org/en-US/docs/Web/API/WebGPU_API) (@matthargett?)
- [ ] [WebXR Device API 🧪](https://developer.mozilla.org/en-US/docs/Web/API/WebXR_Device_API) (@matthargett?)

**Comment by matthargett** [comment]
great stuff! there are three modules I need for my product: WebPayment API, WebXR API, and WebGPU API. I'm excited to contribute these to rnxkit, but wanted to make a note here for visibility in case people want to coordinate/split the work.

**Comment by giacomocerquone** [comment]
> https://github.com/microsoft/rnx-kit/tree/main/incubator/%40react-native-webapis/types

This is really nice. Sadly the posted link of the original issue redirects to a 404, can we get it back somehow please?

**Comment by kelset** [comment]
@giacomocerquone ooops I forgot to update the post, we did end up creating the tool (this https://github.com/microsoft/rnx-kit/blob/main/incubator/%40react-native-webapis/scan ) we meant to have for data-driven decisions, so I've updated the top post to reflect that ;)
